aten/src/ATen/native/Blas.cpp

```diff
@@ -137,6 +140,110 @@ Tensor dot(const Tensor &self, const Tensor &other){
   });
 }
 
+// this is to port LDA_COND from the TH codebase
+// #define LDA_COND(M, N, LDA) ((N) == 1 || (LDA) >= THMax(1, (M)))
+// lda_cond function in this file is not used
+// because it requires `lda > std::max<int64_t>(1L, m)`
+// instead of `>=` as defined in the TH implementation
+constexpr inline bool lda_cond_addr(int64_t m, int64_t n, int64_t lda) {
+  return n == 1 || lda >= std::max<int64_t>(1L, m);
+}
+
+template<typename scalar_t>
+Tensor &addr_impl_cpu(Tensor &result, const Tensor &self,
+                    const Tensor& vec1, const Tensor& vec2,
+                    scalar_t beta, scalar_t alpha) {
+  if (&result != &self) {
+    at::native::resize_as_(result, self);
+    at::native::copy_(result, self);
+  }
+  if (beta == 0.0) {
+    at::native::zero_(result);
```

Comment:
This might happen after result was just copied to. 

Also: why would this fill with zero in this case?

Reply:
I referred the `cuda` implementation
https://github.com/pytorch/pytorch/blob/105132b8915d40528257e6a73be7fd3572756f9c/aten/src/ATen/native/cuda/LinearAlgebra.cu#L188-L194

Below is the math formular for `addr`
![](https://render.githubusercontent.com/render/math?math=\text{out}%20=%20\beta\%20\text{input}%20%20\%2B%20\alpha\%20(\text{vec1}%20\otimes%20\text{vec2}))
The reason might be canceling the result of the copy self to result.
The logic here could be enhanced by not copying self to result if input `beta` is zero.